src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs

```diff
@@ -127,61 +127,121 @@ public CrpgAgentApplyDamageModel(CrpgConstants constants)
         return finalDamage;
     }
 
-    public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman,bool isMissile)
+    public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman, bool isMissile)
     {
         float result = 1f;
-        switch (bodyPart)
-        {
-            case BoneBodyPartType.None:
-                result = 1f;
-                break;
-            case BoneBodyPartType.Head:
-                switch (type)
-                {
-                    case DamageTypes.Invalid:
-                        result = 2f;
-                        break;
-                    case DamageTypes.Cut:
-                        result = 1.2f;
-                        break;
-                    case DamageTypes.Pierce:
-                        result = !isHuman ? 1.2f : 1.6f;
-                        break;
-                    case DamageTypes.Blunt:
-                        result = 1.2f;
-                        break;
-                }
 
-                break;
-            case BoneBodyPartType.Neck:
-                switch (type)
-                {
-                    case DamageTypes.Invalid:
-                        result = 2f;
-                        break;
-                    case DamageTypes.Cut:
-                        result = 1.2f;
-                        break;
-                    case DamageTypes.Pierce:
-                        result = !isHuman ? 1.2f : 1.6f;
-                        break;
-                    case DamageTypes.Blunt:
-                        result = 1.2f;
-                        break;
-                }
-
-                break;
-            case BoneBodyPartType.Chest:
-            case BoneBodyPartType.Abdomen:
-            case BoneBodyPartType.ShoulderLeft:
-            case BoneBodyPartType.ShoulderRight:
-            case BoneBodyPartType.ArmLeft:
-            case BoneBodyPartType.ArmRight:
-                result = !isHuman ? 0.8f : 1f;
-                break;
-            case BoneBodyPartType.Legs:
-                result = 0.8f;
-                break;
+        if (isMissile)
+        {
+            switch (bodyPart)
+            {
+                case BoneBodyPartType.None:
+                    result = 1f;
+                    break;
```

Comment:
The code remains insanely nested.
What i recommend : 
CalculateMeleeDamageMultiplierForHumanBodyPart
CalculateMeleeDamageMultiplierForNonHumanBodyPart
CalculateRangedDamageMultiplierForHumanBodyPart
CalculateRangedDamageMultiplierForNonHumanBodyPart

Then 
If(IsMissile)
{
   return IsHuman ? CalculateRangedDamageMultiplierForHumanBodyPart(ArgumentsGoHere) : CalculateRangedDamageMultiplierForNonHumanBodyPart(ArgumentsGoHere)
}
else
{
 return IsHuman ? CalculateMeleeDamageMultiplierForHumanBodyPart(ArgumentsGoHere) : CalculateMeleeDamageMultiplierForNonHumanBodyPart(ArgumentsGoHere)
}

this should kill any form of nesting

Reply:
Attempted 2nd draft, checking it is what you meant before continuing
I feel like it is wrong still, would add 4 more methods, increasing code, but would make
"public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman, bool isMissile)" method short

Reply:
Unless you also mean to rewrite the TW base code with lots of switches etc?

Reply:
> Attempted 2nd draft, checking it is what you meant before continuing I feel like it is wrong still, would add 4 more methods, increasing code, but would make "public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart, DamageTypes type, bool isHuman, bool isMissile)" method short

Don't worry , we don't have to pay any extra for the extra lines of code.  
You spend 20% of the time writing code , and 80% trying to figure what the hell it means when you come back to it 6 month later. Good code is code that you can understand as you read it. If you have to follow nested condition and remember everytime what branch you are in it makes it harder for you to read. Here's a good short video on the subject
https://www.youtube.com/watch?v=CFRhGnuXG-4

Reply:
ok, updated and finished, + cleaned it up a little